tests/aws_test_harness.h

```diff
@@ -0,0 +1,177 @@
+#ifndef AWS_TEST_HARNESS_H_
+#define AWS_TEST_HARNESS_H_
+/*
+* Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+#include <aws/common/common.h>
+
+#include <stdio.h>
+#include <assert.h>
+#include <stdarg.h>
+#include <stdlib.h>
+#include <string.h>
+
+struct memory_test_config {
+    void *(*mem_acquire)(struct aws_allocator *config, size_t size);
+    void(*mem_release)(struct aws_allocator *config, void *ptr);
+    size_t allocated;
+    size_t freed;
+};
+
+struct memory_test_tracker {
+    size_t size;
+    void *blob;
+};
+
+static void *mem_acquire_malloc(struct aws_allocator *config, size_t size) {
+    struct memory_test_config *test_config = (struct memory_test_config *)config;
+    test_config->allocated += size;
+    struct memory_test_tracker *memory = (struct memory_test_tracker *) malloc(size + sizeof(struct memory_test_tracker));
+    memory->size = size;
+    memory->blob = memory + sizeof(struct memory_test_tracker);
+    return memory->blob;
+}
+
+static void mem_release_free(struct aws_allocator *config, void *ptr) {
+    struct memory_test_config *test_config = (struct memory_test_config *)config;
+
+    struct memory_test_tracker *memory = (struct memory_test_tracker *) (ptr - sizeof(struct memory_test_tracker));
```

Comment:
Pointer arithmetic on `void *` is not allowed by C99. GCC accepts it as an extension; turn on a higher warning level if it didn't flag this.

Reply:
it's at the highest level, maybe it's because that code isn't being hit yet.